Source/LibationFileManager/Templates.cs

```diff
@@ -2,451 +2,347 @@
-			#endregion
-
-			#region to file name
-			/// <summary>USES LIVE CONFIGURATION VALUES</summary>
-			public string GetFilename(LibraryBookDto libraryBookDto, string baseDir = null)
-				=> getFileNamingTemplate(libraryBookDto, Configuration.Instance.FolderTemplate, baseDir ?? AudibleFileStorage.BooksDirectory, null, Configuration.Instance.ReplacementCharacters)
-				.GetFilePath(string.Empty);
-			#endregion
+			return lbConditions;
 		}
 
-		public class FileTemplate : Templates
+		private static PropertyTagClass<LibraryBookDto> GetFilePropertyTags()
 		{
-			public override string Name => "File Template";
-			public override string Description => Configuration.GetDescription(nameof(Configuration.FileTemplate));
-			public override string DefaultTemplate { get; } = "<title> [<id>]";
-			protected override bool IsChapterized { get; } = false;
+			PropertyTagClass<LibraryBookDto> lbProperties = new();
+			lbProperties.RegisterProperty(TemplateTags.Id, lb => lb.AudibleProductId);
+			lbProperties.RegisterProperty(TemplateTags.Title, lb => lb.Title, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.TitleShort, lb => lb.Title.IndexOf(':') < 1 ? lb.Title : lb.Title.Substring(0, lb.Title.IndexOf(':')), StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Author, lb => lb.AuthorNames, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.FirstAuthor, lb => lb.FirstAuthor, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Narrator, lb => lb.NarratorNames, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.FirstNarrator, lb => lb.FirstNarrator, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Series, lb => lb.SeriesName, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.SeriesNumber, lb => lb.SeriesNumber, IntegerFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Language, lb => lb.Language, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.LanguageShort, lb => getLanguageShort(lb.Language), StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Bitrate, lb => lb.BitRate, IntegerFormatter); 
+			lbProperties.RegisterProperty(TemplateTags.SampleRate, lb => lb.SampleRate, IntegerFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Channels, lb => lb.Channels, IntegerFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Account, lb => lb.Account, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.Locale, lb => lb.Locale, StringFormatter);
+			lbProperties.RegisterProperty(TemplateTags.YearPublished, lb => lb.YearPublished, IntegerFormatter);
+			lbProperties.RegisterProperty(TemplateTags.DatePublished, lb => lb.DatePublished, DateTimeFormatter);
+			lbProperties.RegisterProperty(TemplateTags.DateAdded, lb => lb.DateAdded, DateTimeFormatter);
+			lbProperties.RegisterProperty(TemplateTags.FileDate, lb => lb.FileDate, DateTimeFormatter);
+			return lbProperties;
+		}
```

Comment:
> RegisterProperty: Where is this called?

Here.

Reply:
> Large diffs are not rendered by default.

Thanks, github. This is why I couldn't find it. I'd thought: he either did something very clever, or **too** clever

Reply:
> or **too** clever

I try not to do too much of that. Doing something really clever can very fun/satisfying, but then I remember this quote:

> Debugging is twice as hard as writing the code in the first place. Therefore, if you write the code as cleverly as possible, you are, by definition, not smart enough to debug it.

Reply:
> Debugging is twice as hard as writing the code in the first place. Therefore, if you write the code as cleverly as possible, you are, by definition, not smart enough to debug it.

That's a good thing to remember. Another favorite:

> Always code as if the guy who ends up maintaining your code will be a violent psychopath who knows where you live.

The problem here is: that coder was me 4 months ago. He lives 4 months ago where I can't reach him. Fortunately, I know how to ruin that asshole's future.